torch/sparse/semi_structured.py

```diff
@@ -404,6 +413,24 @@ def from_dense(
             requires_grad=original_tensor.requires_grad,
         )
 
+    @classmethod
+    def from_dense_fast(cls, original_tensor : torch.Tensor, algorithm="") -> "SparseSemiStructuredTensor":
```

Comment:
Maybe `from_dense_sort` or so? Fast isn't quite enough right, because the shuffle is also in there.

Reply:
Maybe `from_dense` and `prune_dense` ?

`from_dense` takes a dense matrices with 0s in the right place
`prune_dense`` takes a dense matrix and will put the 0s in the right place as part of the process.

Reply:
@cpuhrsch How about `prune_dense_static_sort` or `prune_dense_sorting_network` then?

Reply:
Hm, why `network`? The decomposition should also help a lot.

Reply:
It's how we do the sort https://en.wikipedia.org/wiki/Sorting_network